src/vkscript/command_parser.cc

```diff
@@ -615,6 +615,10 @@ Result CommandParser::ProcessSSBO() {
     if (!r.IsSuccess())
       return r;
 
+    // Resize the buffer so we'll know the max size
+    if (buf->ElementCount() < values.size())
+      buf->SetElementCount(static_cast<uint32_t>(values.size()));
```

Comment:
I don't think this is quite correct. You need to take the offset into account as well as the values size. If we set 2 values at offset 50, we need to make sure the buffer is 50+(values.size()) bytes, which also doesn't necessarily match up to element counts ....

Reply:
Yes that's right. How about calling SetDataWithOffset?